src/components/ModularTable/ModularTable.tsx

```diff
@@ -0,0 +1,84 @@
+import React from "react";
+import PropTypes from "prop-types";
+import { useTable } from "react-table";
+import type { Column } from "react-table";
+
+import Table from "../Table";
+import TableRow from "../TableRow";
+import TableHeader from "../TableHeader";
+import TableCell from "../TableCell";
+
+export type Props<D extends Record<string, unknown>> = {
+  columns: Column<D>[];
+  data: D[];
+};
+
+function ModularTable({
+  data,
+  columns,
+}: Props<Record<string, unknown>>): JSX.Element {
+  const {
+    getTableProps,
+    getTableBodyProps,
+    headerGroups,
+    rows,
+    prepareRow,
+  } = useTable({ columns, data });
+
+  return (
+    <Table {...getTableProps()}>
+      <thead>
+        {headerGroups.map((headerGroup) => (
+          <TableRow {...headerGroup.getHeaderGroupProps()}>
+            {headerGroup.headers.map((column) => (
+              <TableHeader
+                {...column.getHeaderProps([
+                  {
+                    className: column.className,
+                  },
+                ])}
+              >
+                {column.render("Header")}
+              </TableHeader>
+            ))}
+          </TableRow>
+        ))}
+      </thead>
+      <tbody {...getTableBodyProps()}>
+        {rows.map((row) => {
+          prepareRow(row);
```

Comment:
This is interesting, I had to look up why it was necessary (they have great docs btw...)

Could you add a comment here just to say what this is and that this will need to be updated if we're doing pagination?

Reply:
Sure, will do.